### ISSUE TITLE
nsolve does not return sympy objects

### ISSUE THREAD

**Issue by Krastanov** [sympy/sympy#6092]
```
The matrices that are returned are sympy.mpmath.matrices.matrices.matrix.
Number are returned as float (build-in).
```

Original issue for #6092: http://code.google.com/p/sympy/issues/detail?id=2993
Original author: https://code.google.com/u/100157245271348669141/

**Comment by cbm755** [comment]
For scalars this is fixed by #11888.

@asmeurer the OP mentions matrices: I don't know how to make nsolve return a matrix...  Is that still relevant?  Anyway, thought I should double-check before I close this.

**Comment by asmeurer** [comment]
It does return matrices now, when passes a system.
